### PR TITLE
auth: ignore cryptokeys in presigned zones

### LINKED ISSUE #9489: Confusion when DNSSEC key(s) exist for PRESIGNED zone
- Program: Authoritative
 - Issue type: Bug report

### Short description
If a PRESIGNED DNSSEC secured zone has key(s) attached to it, they are used to sign the zone. This is not indicated by `pdnsutil show-zone` output which shows only the AXFR'd keys from primary. To figure out you have such key(s) one has to do `pdnsutil list-keys`.

### Environment
 - Operating system: Debian 9.13
 - Software version:  PowerDNS Authoritative Server 4.2.2
 - Software source: PowerDNS

### Steps to reproduce
1. Add keys for a zone
2. Configure zone as slave
3. AXFR presigned zone

### Expected behaviour
Not sure what should happen, but at least `pdnsutil show-zone` should indicate what the keys are that are used.

### Actual behaviour
PowerDNS tools indicate that zone is using presigned data, and correctly show the presigned key from master. Yet the server uses existing DNSSEC keys to live sign the record data.

### Other information
Supporting live signing on slaves could be used to avoid an extra unsigned master, and could be used to support NSEC3 NARROW slaving?

### PR DESCRIPTION
**Pull request by mind04**
### Short description
Fixes #9489

### Checklist
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by mnordhoff** [comment]
> Fixes #9492 

That should be fixes #9489, right?

**Comment by mind04** [comment]
oops, updated

**Comment by cmouse** [comment]
Can we add also feature that will warn about this with `pdnsutil show-zone`? Something like `cryptokeys found for presigned zone`?

File: pdns/tcpreceiver.cc [hunk 1]
```diff
@@ -609,70 +609,73 @@ int TCPNameserver::doAXFR(const DNSName &target, std::unique_ptr<DNSPacket>& q,
...
+    for (auto const& value : entryPoints)
+      entryPointIds.insert(value.second.id);
+
+    for(const DNSSECKeeper::keyset_t::value_type& value :  keys) {
+      if (!value.second.published) {
+        continue;
+      }
+      zrr.dr.d_type = QType::DNSKEY;
+      zrr.dr.d_content = std::make_shared<DNSKEYRecordContent>(value.first.getDNSKEY());
+      DNSName keyname = NSEC3Zone ? DNSName(toBase32Hex(hashQNameWithSalt(ns3pr, zrr.dr.d_name))) : zrr.dr.d_name;
```
Comment (cmouse) [inline pdns/tcpreceiver.cc:637]: `const DNSName&`

File: pdns/tcpreceiver.cc [hunk 2]
```diff
@@ -609,70 +609,73 @@ int TCPNameserver::doAXFR(const DNSName &target, std::unique_ptr<DNSPacket>& q,
...
+      NSECXEntry& ne = nsecxrepo[keyname];
```
Comment (cmouse) [inline pdns/tcpreceiver.cc:638]: `const NSECXEntry&`

**Review by cmouse — CHANGES_REQUESTED** [review]

**Pull request merged**